juju-tutorials/how-to-deploy-openstakc-on-openstack.md

```diff
@@ -0,0 +1,371 @@
+
+2. The last thing is to add a new OpenStack cloud to Juju, create new credentials and create a controller. 
+   See the [OpenStack-cloud][openstack-cloud] page for more information.
+   
+   1. Add OpenStack as a cloud to Juju.
+      ```bash
+      juju add-cloud
+      ```
+      [note="cloud configuration example"]
+      ```yaml
+      defined: public
+      type: openstack
+      auth-types: [userpass]
+      endpoint: https://10.5.0.8:5000/v3
+      credential-count: 1
+      regions:
+         RegionOne: {}
+      ca-credentials:
+      - |
+        -----BEGIN CERTIFICATE-----
+        <your_certificate>
+        -----END CERTIFICATE-----
+
+      ```
+      [/note]
+   2. Add credentials to OpenStack cloud.
+      ```bash
+      juju add-credential <cloud_name>
+      ```
+      [note="credentials example]
+      ```yaml
+      <cloud_name>:
+        <credentials name>:
+          auth-type: userpass
+          username: admin
+          password: <user_password>
+          domain-name: ""
+          project-domain-name: admin_domain
+          tenant-name: admin
+          user-domain-name: admin_domain
+          version: "3"
+      ```
+      [\note]
+   3. Allow multiple units to land in 3 nova-compute units (edit disk, ram and cpu ratio).
+      ```bash
+      juju config nova-cloud-controller disk-allocation-ratio=10
+      juju config nova-cloud-controller ram-allocation-ratio=1
+      juju config nova-cloud-controller cpu-allocation-ratio=4
+      ```
+   4. It's necessary to create metadata for initializing a Juju cloud environment (creating the controller).
+      ```bash
+      IMG_ID=$(openstack image list --name focal -c ID -f value)
+      juju metadata generate-image -d ./simplestreams -i $IMG_ID -s focal -r $OS_REGION_NAME -u $OS_AUTH_URL
+      ```
+   5. Bootstrap the controller.
+      ```bash
+      EXT_NET=$(openstack network list --name external_net -c ID -f value)
+      INT_NET=$(openstack network list --name internal_net -c ID -f value)
+      juju bootstrap --metadata-source ./simplestreams  --model-default external-network=$EXT_NET --model-default network=$INT_NET --model-default use-floating-ip=true --model-default use-default-secgroup=true --config default-series=focal <cloud_name> <controller_name>
+      ```
+      [note status="Proxy"]
+      If a proxy is required on the servers where you deploy it, also use 
+      `--model-default program http-proxy=<proxy_ip>:<proxy_port> https-proxy=<proxy_ip>:<proxy_port>`. 
+      [/note]
+
+### Kubernetes
+
+Deploy CDK (link to CDK) on top of OpenStack using charm-openstack-integrator. The integrator will act as a proxy 
+between Kubernetes and OpenStack, and it will provide a control interface (e.g. creating persistent volumes).
+
+```bash
+juju deploy charmed-kubernetes --overlay openstack-overlay.yaml --trust
+```
+
+[note status="OpenStack credentials"]
+Before configuration `openstack-integrator` load the OpenStack credentials as environment variables. 
```

Comment:
Suggestion: "In order to configure the `openstack-integrator` application, the OpenStack credentials need to be loaded as environment variables."